src/components/Button/Button.tsx

```diff
@@ -74,11 +86,36 @@ class Button extends UIComponent<IButtonProps, any> {
     classes,
     rest,
   }: IRenderResultConfig<IButtonProps>): ReactNode {
-    const { children, content } = this.props
+    const { children, content, icon, iconPosition, type } = this.props
+    const primary = type === 'primary'
+
+    const getContent = (): ReactNode => {
+      if (childrenExist(children)) {
+        return children
+      }
+
+      const renderedContent = [
+        content && (
+          <span key="btn-content" className={classes.span}>
+            {content}
+          </span>
+        ),
+        icon && (
+          <Icon
+            key="btn-icon"
+            name={icon}
+            fitted
+            color={primary ? 'white' : 'black'}
+          />
```

Comment:
the problem is not with `Shorthand API` because there I can achieve that by adding space to the span instead by styling it however I want (it's an implementation detail); however, in `Children API` it's not an implementation detail as it's part of the API the user will have to use to achieve the same styling.

tl;dr
Your comment still holds and it's basically agreeing with my suggestion in the previous comment

> margin="left|right": for adding space to the left or right depending on where it's positioned relative to the text; something like:
```jsx
<Button type="primary" icon>
  <Icon name="book" color="white" margin="right" />
  <span>Click me left</span>
</Button>
<Button type="secondary" icon>
  <span>Click me right</span>
  <Icon name="coffee" margin="left" />
</Button>
```

I believe it's the best option right now to add that margin to the `Icon` component;
@levithomason what are your thoughts?

Reply:
@mnajdova since she hit the same issue

Reply:
unblocked by adding `xSpacing` prop to Icon component in #22